ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -238,11 +238,16 @@ wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
 
     int ret = s3->wtApi->config_get(s3->wtApi, NULL, config, "verbose", &v);
 
-    if (ret == WT_NOTFOUND) {
+    // If a verbose level is not found, it will set the level to -3 (Error).
```

Comment:
What do you think of the following in terms of readibiliy?
```
    if (ret == 0) {
        if (v.val < -3 || v.val > 1)
            return EINVAL;
        s3->verbose = v.val;
    } else if (ret == WT_NOTFOUND)
        s3->verbose = -3;

    return ret;
```

Reply:
We can't pull out the `return ret` as there's more code that follows. I did just realize that we do a memory allocation prior to this piece of code, so we need to free the `s3` structure before any early return.

Reply:
Thanks @tammybailey, my bad, I did not expand the window and thought it was the end of the function. The `free` is a real issue, good finding.

Reply:
I have applied the following suggestion from @tammybailey  

`if (ret == 0 && v.val >= -3 && v.val <= 1)
    s3->verbose = v.val;
else if(ret == WT_NOTFOUND)
    s3->verbose = -3;
else {
    free(s3);
    return (ret != 0 ? ret : EINVAL);
}`